ext/pdo_pgsql/pdo_pgsql.c

```diff
@@ -53,6 +58,90 @@ zend_module_entry pdo_pgsql_module_entry = {
 ZEND_GET_MODULE(pdo_pgsql)
 #endif
 
+/* Escape an identifier for insertion into a text field	*/
+PHP_METHOD(PdoPgsql, escapeIdentifier)
+{
+	zend_string *from = NULL;
+	char *tmp;
+	pdo_dbh_t *dbh;
+	pdo_pgsql_db_handle *H;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "S", &from) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	dbh = Z_PDO_DBH_P(ZEND_THIS);
+	PDO_CONSTRUCT_CHECK;
+	PDO_DBH_CLEAR_ERR();
+
+	/* Obtain db Handle */
+	H = (pdo_pgsql_db_handle *)dbh->driver_data;
+
+	tmp = PQescapeIdentifier(H->server, ZSTR_VAL(from), ZSTR_LEN(from));
+	if (!tmp) {
+		// TODO - exception
+		php_error_docref(NULL, E_WARNING,"Failed to escape identifier");
```

Comment:
The TODO maybe? Although I imagine that would be a behavioural change that can be addressed in a follow-up possibly.

Reply:
Ah, I intentionally kept this todo so that it provokes discussion :) Curently, ext/pgsql uses the same warning and the other problem is that I'm not sure (even by reading https://www.postgresql.org/docs/current/libpq-exec.html) what kind of errors can happen.

Reply:
@devnexen maybe you have more insight into this?